### PR TITLE
feat: hand off and hand on work between main and a worktree

### LINKED ISSUE #646: feat: hand off and hand on work between main and a worktree
### Problem or use case

Work often starts on the main worktree and later needs a child worktree, or needs to come back to main. That move is not a first-class action today.

### Proposed solution

Move current work from the main worktree into a new worktree, and bring work from a child worktree back onto main.

### PR DESCRIPTION
**Pull request by leynier**
## Summary

Adds first-class **Hand Off** (main → child) and **Hand On** (child → main) actions so current git work can move between the main worktree and a linked child without a general worktree manager.

- From the main worktree: Hand Off stashes uncommitted changes (including untracked files), creates a linked child workspace, and applies the work there. A new branch name creates a branch from current HEAD; using the current feature branch name moves that branch off main.
- From a child worktree: Hand On brings the child's branch and uncommitted changes onto main, then removes the child worktree and workspace record. The branch is kept. Main must be clean.
- Desktop: workspace context menu + command palette. CLI: `alera workspace hand-off` / `alera workspace hand-on`. Host RPCs: `workspace.handOff` / `workspace.handOn` (additive, no protocol version bump).

Closes #646

### AC coverage

1. Main → child hand off: first-class action (context menu, command palette, CLI, host RPC).
2. Child → main hand on: first-class action (same surfaces).
3. Focused: reuses managed workspace create/remove; no general worktree manager.
4. Focused tests for both directions (Rust git + managed workspace, Dart client/service/controller, dialog/widget).
5. Demo: BOX required after HARD. Author prepared the script below; QA/Alera should record on the box computer, not miniPC.
6. CI + QA HARD before merge (this PR does not merge).

## BOX demo script (record after HARD)

1. Open a git project on its main workspace. Make a tracked edit and add an untracked file.
2. Right-click the main workspace → **Hand Off**. Enter a new branch (for example `feat/646-demo`) and confirm.
3. Confirm the child workspace is selected, has the branch and both files, and main is clean on its previous branch.
4. Add another untracked file in the child. Right-click the child → **Hand On**. Confirm.
5. Confirm main now has the child's branch and files, and the child workspace is gone.

## Testing

- [x] Added focused tests for both directions
- [x] `dart format` on changed Dart files
- [x] `dart run tool/quality/check_max_lines.dart`
- [x] `flutter analyze`
- [x] `flutter test` `test/unit/workspace_service_test.dart` `test/unit/workbench_controller_test.dart` `test/unit/runtime_repositories_test.dart` `test/unit/project_workbench_workspace_actions_test.dart` `test/unit/keyboard_command_palette_test.dart` `test/unit/keybinding_resolver_test.dart` `test/widget/workspace_hand_off_dialog_test.dart` `test/widget/keyboard_command_dispatcher_test.dart`
- [x] `cargo test -p alera-core worktree_handoff`
- [x] `cargo test -p alera-cli managed_workspace_handoff`
- [x] `cargo fmt --all --manifest-path rust/Cargo.toml`
- [x] `cargo clippy -p alera-core --all-targets --manifest-path rust/Cargo.toml -- -D warnings`
- [x] `cargo clippy -p alera-cli --all-targets --manifest-path rust/Cargo.toml -- -D warnings`

## Notes

- Hand on refuses when main has local changes.
- Hand off of the default branch requires a new branch name.
- Worktree setup on the new child follows the existing deferred Setup terminal path on desktop.
- Docs updated: `docs/architecture.md`, `skills/alera-cli/SKILL.md`. `AGENTS.md` unchanged (CLI skill covers the operator surface).

### CODE REVIEW

**Comment by leynier** [comment]
## BOX demo (post-HARD)

Tip `01b6061c` — Hand Off main→child (`feat/646-demo`) then Hand On child→main.

- Video: https://github.com/leynier/alera/releases/download/alera-pr705-demo/pr705-recording_demo.mp4
- Duration ~1m36s; recorded on Alera box (not miniPC)

Clear to squash-merge.

File: rust/alera-cli/src/managed_workspace_handoff.rs [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+        .await?
+        .ok_or_else(|| anyhow!("Project not found: {}", child.project_id))?;
+    let mut main = find_main_workspace(store, &project.id).await?;
+    let branch = child
+        .branch
+        .as_deref()
+        .map(str::trim)
+        .filter(|value| !value.is_empty())
+        .ok_or_else(|| anyhow!("Child workspace has no branch to bring onto main"))?
+        .to_string();
```
[the remark this answers is not in the document]
Comment (leynier) [reply rust/alera-cli/src/managed_workspace_handoff.rs:117]: EM Option B: OOS for #646 locked AC (first-class hand off/on + tests + demo). Live HEAD vs workspace metadata drift is an edge case — tracked as follow-up https://github.com/leynier/alera/issues/707. No tip change / no re-HARD on this PR.
